Python/Product/PythonTools/visualstudio_py_debugger.py

```diff
@@ -2071,22 +2101,11 @@ def report_exception(frame, exc_info, tid, break_type):
             'message': data['message'],
         }
 
-    with _SendLockCtx:
-        if RICH_EXCEPTIONS:
-            write_bytes(conn, EXC2)
-            write_int(conn, tid)
-            write_int(conn, len(data))
-            for key, value in data.items():
-                write_string(conn, key)
-                write_string(conn, str(value))
-        else:
-            # Old message is fixed format. If RichExceptions is not passed in
-            # debug options, we'll send this format.
-            write_bytes(conn, EXCP)
-            write_string(conn, str(data['typename']))
-            write_int(conn, tid)
-            write_int(conn, 1 if 'breaktype' in data else 0)
-            write_string(conn, str(data['message']))
+    send_debug_event(
+        name='legacyException',
+        threadId=tid,
+        data=[{'key':k, 'val':str(v)} for k, v in data.items()],
```

Comment:
Guessing it's minimal change right now. Probably not worth doing anything about it

---
In reply to: [107989912](https://github.com/Microsoft/PTVS/pull/2323#discussion_r107989912) [](ancestors = 107989912)